File: app/src/main/java/com/example/pizzeria/DAOS/DAOKebab.java
package com.example.pizzeria.DAOS;

import com.example.pizzeria.ConexionBD.DBHelper;
import com.example.pizzeria.Enums.TipoIngredientes;
import com.example.pizzeria.Recursos.Kebab;

import java.util.ArrayList;

public class DAOKebab {

    private static ArrayList<Kebab> lista;
    private static DAOKebab dao;
    private DBHelper dbHelper;

    public static DAOKebab getInstance(){

        if (dao==null)
            dao=new DAOKebab();
        if(lista==null)
            lista=new ArrayList<Kebab>();
        return dao;
    }

    public void setDbHelper(DBHelper dbHelper) {
        this.dbHelper = dbHelper;
    }

    public ArrayList<Kebab> getLista(){
        return lista;
    }

    public Kebab getKebab(Kebab kebab){
        if(DAOKebab.getInstance().getLista().contains(kebab)){
            return kebab;
        }
        else return null;
    }
    public Boolean agregarKebab(Kebab kebab,String usuario){
        lista.add(kebab);
        dbHelper.insertarKebab(kebab, usuario);
        int idKebab=dbHelper.idKebab();
        for (TipoIngredientes ingrediente: kebab.getIngredientes()) {
            dbHelper.insertarIgrediente(ingrediente.ordinal(),idKebab);
        }

        return true;
    }

    public Boolean eliminarultimoKebab(){
        lista.remove(lista.size()-1);
        return true;
    }

    public int getSizeLista(){
        return lista.size();
    }

    public boolean eleminarFavorito(String usuario){
        dbHelper.limpiarFavorito(usuario);
        for(int ind=0;ind<lista.size();ind++){
            lista.get(ind).setFavorito(false);
        }
         return true;
    }
}

File: app/src/main/java/com/example/pizzeria/Kotlin/DAOKebab.kt
package com.example.pizzeria.Kotlin

import com.example.pizzeria.ConexionBD.DBHelper
import com.example.pizzeria.Recursos.Kebab

class DAOKebab {
    private var dbHelper: DBHelper? = null
    fun setDbHelper(dbHelper: DBHelper?) {
        this.dbHelper = dbHelper
    }

    val lista: ArrayList<Kebab>?
        get() = Companion.lista

    fun getKebab(kebab: Kebab): Kebab? {
        return if (instance!!.lista!!.contains(kebab)) {
            kebab
        } else null
    }

    fun agregarKebab(kebab: Kebab, usuario: String?): Boolean {
        Companion.lista!!.add(kebab)
        dbHelper!!.insertarKebab(kebab, usuario)
        val idKebab = dbHelper!!.idKebab()
        for (ingrediente in kebab.ingredientes) {
            dbHelper!!.insertarIgrediente(ingrediente.ordinal, idKebab)
        }
        return true
    }

    fun eliminarultimoKebab(): Boolean {
        Companion.lista!!.removeAt(Companion.lista!!.size - 1)
        return true
    }

    val sizeLista: Int
        get() = Companion.lista!!.size

    fun eleminarFavorito(usuario: String?): Boolean {
        dbHelper!!.limpiarFavorito(usuario)
        for (ind in Companion.lista!!.indices) {
            Companion.lista!![ind].favorito = false
        }
        return true
    }

    companion object {
        private var lista: ArrayList<Kebab>? = null
        private var dao: DAOKebab? = null
        val instance: DAOKebab?
            get() {
                if (dao == null) dao = DAOKebab()
                if (lista == null) lista = ArrayList()
                return dao
            }
    }
}